src/txn-wal/src/lib.rs

```diff
@@ -282,16 +283,17 @@ pub struct TxnsCodecDefault;
 
 impl TxnsCodec for TxnsCodecDefault {
     type Key = ShardId;
-    type Val = Vec<u8>;
+    type Val = Bytes;
     fn schemas() -> (<Self::Key as Codec>::Schema, <Self::Val as Codec>::Schema) {
         (ShardIdSchema, VecU8Schema)
     }
     fn encode(e: TxnsEntry) -> (Self::Key, Self::Val) {
         match e {
-            TxnsEntry::Register(data_id, ts) => (data_id, ts.to_vec()),
+            TxnsEntry::Register(data_id, ts) => (data_id, Bytes::from(ts.to_vec())),
             TxnsEntry::Append(data_id, ts, batch) => {
                 // Put the ts at the end to let decode truncate it off.
-                (data_id, batch.into_iter().chain(ts).collect())
+                let batch = batch.into_iter().chain(ts).collect::<Vec<_>>();
```

Comment:
This might be slower than the previous impl. The older version may have been able to extend `batch`, but we definitely have to clone it.